tutorial/under-the-hood/Compilation.md

```diff
@@ -368,14 +368,15 @@ TODO: ...
 
 ## Variables
 
-This is not implemented yet, but this is an example of what they
-will look like.
+The `x:variable` element in the XSpec namespace defines an XSpec variable. Any number of `x:variable` elements can occur as a child of `x:description`, or before or after `x:context`, `x:call`, `x:apply` (not implemented yet), or the first `x:expect`. XSpec variables can be referenced in XPath expressions, such as in `@select` and `@test` attributes.
```

Comment:
> the first `x:expect`

Is this `the first` necessary?

Reply:
Good catch. That phrase is not necessary and I'll remove it. I must have written that when I was debating with myself about x:variable interspersed with x:expect (see "A few questions and comments" in this PR description at the top). In this PR, I did not make `x:variable` valid in the schema as a child of `x:pending` but I did make `x:variable` work interspersed with (any number of) `x:expect` elements. The file `test/xspec-variable.xspec` has a variable named `myv:after_expect` that is defined between two sibling `x:expect` elements.

Reply:
Just to show that the second `x:expect` element is no different from the first in that it can have a variable either before or after it, I added a test case for a variable after the second expect. The changes on this point are b6fcabb26792073092b958e9e93200ef0d10ff44 and b3c6430f84525e08b398e3fbd1d0e9461f1d60fb.